src/openms/include/OpenMS/FORMAT/MSstatsFile.h

```diff
@@ -65,16 +63,89 @@ namespace OpenMS
+                      const ExperimentalDesign& design,
+                      const StringList& reannotate_filenames,
+                      const bool is_isotope_label_type,
+                      const String& bioreplicate,
+                      const String& condition,
+                      const String& retention_time_summarization_method);
+        
+        // store isobaric experiment (MSstatsTMT)
+        void storeISO(const String& filename, 
+                      ConsensusMap &consensus_map,
+                      const ExperimentalDesign& design,
+                      const StringList& reannotate_filenames,
+                      const String& bioreplicate,
+                      const String& condition,
+                      const String& mixture,
+                      const String& retention_time_summarization_method);
+    
     private:
+     
+        const String na_string = "NA";
+        
+        // The meta value of the peptide identification which is going to be used for the experimental design link
+        const String meta_value_exp_design_key = "spectra_data";
+
+        /*
+        *  @Brief: Internal function to check if MSstats_BioReplicate and MSstats_Condition in Experimental Design
+        */
+        static void checkConditionLFQ_(const ExperimentalDesign::SampleSection& sampleSection, const String& bioreplicate, const String& condition);
+
+        /*
+         *  @Brief: Internal function to check if MSstats_BioReplicate, MSstats_Condition and MSstats_Mixture in Experimental Design
+         */
+        static void checkConditionISO_(const ExperimentalDesign::SampleSection sampleSection, const String& bioreplicate, const String& condition, const String& mixture);
+
+        /*
+         *  MSstats treats runs differently than OpenMS. In MSstats, runs are an enumeration of (SpectraFilePath, Fraction)
+         *  In OpenMS, a run is split into multiple fractions.
+         *
+         */
+        static void assembleRunMap(
+                std::map< std::pair< String, unsigned>, unsigned> &run_map,
+                const ExperimentalDesign &design)
+        {
+          run_map.clear();
+          const ExperimentalDesign::MSFileSection& msfile_section = design.getMSFileSection();
+          unsigned run_counter = 1;
+
+          for (ExperimentalDesign::MSFileSectionEntry const& r : msfile_section)
+          {
+            std::pair< String, unsigned> tpl = std::make_pair(File::basename(r.path), r.fraction);
+            if (run_map.find(tpl) == run_map.end())
+            {
+              run_map[tpl] = run_counter++;
+            }
+          }
+        }
+
+        bool checkUnorderedContent_(const std::vector< String> &first, const std::vector< String > &second)
+        {
+          const std::set< String > lhs(first.begin(), first.end());
+          const std::set< String > rhs(second.begin(), second.end());
+          return lhs.size() == rhs.size()
```

Comment:
would a simple 
return lhs == rhs 
be enough?

Reply:
Should be enough but I am not sure what the intention of Lukas was here.